src/sqlfluff/core/parser/parsers.py

```diff
@@ -100,6 +100,47 @@ def match(
         return MatchResult.from_unmatched(segments)
 
 
+class MultiStringParser(StringParser):
```

Comment:
Is there some benefit to inheriting from `StringParser`? I notice that:
* In the constructor, we delete one of the parent class' fields (`template`)
* None of the methods use the superclass

Basically, I'm wondering if this class doesn't need to inherit from `StringParser`.

Reply:
There are other methods which are used from StringParser. Though, you are right that the inheritance is odd. I reckon there should be a `BaseParser` perhaps which inherits from `Matchable` and implements the reusable bits of `StringParser` given that it looks to me like all other Parsers inherit from `StringParser`?

Reply:
Had a go at this and keen for feedback.